web/api/web_api_v1.c

```diff
@@ -779,6 +816,7 @@ static struct api_command {
         { "alarms",          0, WEB_CLIENT_ACL_DASHBOARD, web_client_api_request_v1_alarms          },
         { "alarm_log",       0, WEB_CLIENT_ACL_DASHBOARD, web_client_api_request_v1_alarm_log       },
         { "alarm_variables", 0, WEB_CLIENT_ACL_DASHBOARD, web_client_api_request_v1_alarm_variables },
+        { "alarm_counts",    0, WEB_CLIENT_ACL_DASHBOARD, web_client_api_request_v1_alarm_counts    },
```

Comment:
Why `counts`?

```json
{
{net.net:'CLEAR'},
{disk.disk:'REMOVED'},
{apache.requests:'REMOVED'},
{cpu.cpu:'CLEAR'}
}
```

The endpoint returns highest status among all alarms for given context. Where is count part? I dont see any counters

Reply:
This is the name (a bit modified) for the service specified in the issue ticket #6536 by @cakrit . The proposed API url is api/v1/alarmcount. I have changed it to api/v1/alarm_counts just because there are several urls with underscore already (api/v1/alarm_variables and api/v1/alarm_log). Also since api/v1/alarm_counts accepts multiple contexts I though it would be better to have name in plural.

As fas as I understand from the issue the intend of this API is to aggregate several alarms of the same context and provide one value based on several charts. I used maximum for aggregation, but there could be other possibilities like average etc.

So I just used the name proposed in the issue, because I do not see a better options right now.

Reply:
> So I just used the name proposed in the issue, because I do not see a better options right now.

> right now

It is easy to add something, but kind of impossible to change later. So it is better to think before adding.


What does the endpoint return?

Reply:
@cakrit i think the endpoint should be `alarm_status_count` because it returns status.

On the other hand `alarm_count` is more generic and we will be able to add some other count types in the future. Then request should be `/alarm_count?type=status&context=some_context` or just assume that default type is `status`. In that case `alarm_count` is ok.

Reply:
Renamed /api/v1/alarm_counts to '/api/v1/alarm_count'

Reply:
Renaming is ok, this is resolved.